src/containers/Tenant/Query/QuerySettingsDialog/QuerySettingsDialog.tsx

```diff
@@ -0,0 +1,183 @@
+import React from 'react';
+
+import {Dialog, Link as ExternalLink, TextInput} from '@gravity-ui/uikit';
+
+import {
+    selectQueryAction,
+    setQueryAction,
+} from '../../../../store/reducers/queryActions/queryActions';
+import type {
+    IsolationLevel,
+    QueryMode,
+    StatisticsMode,
+    TracingLevel,
+} from '../../../../types/store/query';
+import {cn} from '../../../../utils/cn';
+import {useTypedDispatch, useTypedSelector} from '../../../../utils/hooks';
+import {
+    ISOLATION_LEVELS,
+    QUERY_MODES,
+    STATISTICS_MODES,
+    TRACING_LEVELS,
+} from '../../../../utils/query';
+
+import {QuerySettingsSelect} from './QuerySettingsSelect';
+import {
+    ISOLATION_LEVEL_SELECT_OPTIONS,
+    QUERY_MODE_SELECT_OPTIONS,
+    STATISTICS_MODE_SELECT_OPTIONS,
+    TRACING_LEVEL_SELECT_OPTIONS,
+} from './constants';
+import i18n from './i18n';
+
+import './QuerySettingsDialog.scss';
+
+const b = cn('ydb-query-settings-dialog');
+
+export function QuerySettingsDialog() {
+    const dispatch = useTypedDispatch();
+    const queryAction = useTypedSelector(selectQueryAction);
+    const [queryMode, setQueryMode] = React.useState<QueryMode>(QUERY_MODES.script);
+    const [timeout, setTimeout] = React.useState('60');
+    const [isolationLevel, setIsolationLevel] = React.useState<IsolationLevel>(
```

Comment:
It seems that it will appear more and more different forms in our project. Let's start using https://react-hook-form.com/ for  it.